main.py

```diff
@@ -202,6 +203,11 @@ def upload_cv_upload(person_id, signature):
 
 @app.route('/email_candidate/<int:person_id>', methods=['GET','POST'])
 def email_candidate(person_id):
+    if 'postcode' not in flask.session:
+        # jump out
+        flask.flash("Something went wrong", 'danger')
```

Comment:
A bit aggressive. Again case where this might happen is if, say, somebody tweeted the link.

A success flash saying something more like "Choose constituency to email candidates"